utils/visual/env_utils.py

```diff
@@ -88,30 +88,38 @@ def save_credentials(api_key: str, additional_vars: Optional[Dict[str, Any]] = N
     return 'Credentials saved successfully!'
 
 
-def get_wandb_key() -> Optional[Any]:
+import netrc
+import os
+from typing import Optional
+
+
+def get_wandb_key() -> Optional[str]:
+    """
+    Retrieve the Weights & Biases API key from the environment or ~/.netrc,
```

Comment:
What is `netrc`?

Reply:
The .netrc file contains login information used by the auto-login process. It's a standard used by a lot of companies including W&B